Исправление ошибки 401

AlexandrGerasimov63 opened the pull request:
Ребята, у меня вопросы остались по этой задаче.
Есть ли нормальное решение или это и есть нормальное решение?

AlexandrGerasimov63 committed:
-Исправление ошибки 401

AlexandrGerasimov63 committed:
-Исправление ошибки 401

INextYP commented:
Привет! А если токен вынести в файл с константами и оттуда дергать? Обычно так делаем, дабы избежать дублирования кода и если токен понадобится в других местах

kspshnik commented:
> Привет! А если токен вынести в файл с константами и оттуда дергать? Обычно так делаем, дабы избежать дублирования кода и если токен понадобится в других местах

А ничего, что токен на каждой авторизации новый выдаётся? :)

INextYP commented:
> > Привет! А если токен вынести в файл с константами и оттуда дергать? Обычно так делаем, дабы избежать дублирования кода и если токен понадобится в других местах
> 
> А ничего, что токен на каждой авторизации новый выдаётся? :)

Не так написал.

Я имею ввиду, что можно сделать что-то типа такого:
```js
export const LOCAL_STORAGE_TOKEN_ACCESS = "token_access";

export const getTokenAccess = () => {
  return localStorage.getItem(LOCAL_STORAGE_TOKEN_ACCESS);
};

export const setTokenAccess = (token: string) => {
  return localStorage.setItem(LOCAL_STORAGE_TOKEN_ACCESS, token);
};
```

kspshnik commented:
> Я имею ввиду, что можно сделать что-то типа такого:
> 
> ```js
> export const LOCAL_STORAGE_TOKEN_ACCESS = "token_access";
> 
> export const getTokenAccess = () => {
>   return localStorage.getItem(LOCAL_STORAGE_TOKEN_ACCESS);
> };
> 
> export const setTokenAccess = (token: string) => {
>   return localStorage.setItem(LOCAL_STORAGE_TOKEN_ACCESS, token);
> };
> ```

Не можно, а **НУЖНО**!
Согласен полностью.

AlexandrGerasimov63 committed:
-вынес константу токена
-добавил функции получения и добавления токена, название поменялось с token на access_token
-в userTaskApi добавил prepareHeaders, удалил лишний код
-в system-slice заменил получение токена на функцию

AlexandrGerasimov63 committed:
-вынес константу токена
-добавил функции получения и добавления токена, название поменялось с token на access_token
-в userTaskApi добавил prepareHeaders, удалил лишний код
-в system-slice заменил получение токена на функцию

AlexandrGerasimov63 committed:
Merge branch 'fix/error_401' of github.com:ya-pomogau/frontend into fix/error_401

INextYP commented on src/shared/libs/permissions-names.ts:
Странно, что ты положил эту переменную именно сюда)

INextYP commented on src/services/user-task-api.ts:
```diff
@@ -60,15 +57,10 @@ export const userTasksApi = createApi({
           url: `${role}/tasks/virgin?distance=100000&latitude=${latitude}&longitude=${longitude}`,
           method: 'GET',
-          headers,
+          // headers,
```
Давай не будем оставлять закомментированный код!?)

marussia1996 reviewed (commented):
токен запихни в файл \frontend\src\config\api-config.ts, а файла permissions у нас по-идеи не должно быть, там не актуалььная информация

AlexandrGerasimov63 committed:
-Перенес константу токена
-Удалил закомментированный код

AlexandrGerasimov63 committed:
-Удалил переменную
